frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -785,6 +785,12 @@ export const surveyLogic = kea<surveyLogicType>([
             return [urls.survey(values.survey.id), router.values.searchParams, hashParams]
         },
     })),
+    beforeUnload(({ values }) => ({
+        enabled: (newLocation) => {
+            return values.isEditingSurvey && newLocation?.pathname !== router.values.location.pathname
```

Comment:
the router stuff probably isn't super necessary; really, all we care about if if the user is in an edit session or not.  I just followed the pattern from some of the other `beforeUnload` handlers in our different scenes.

Reply:
Flyby: Beforeunloads are notoriously flakey. We've been getting support requests for the one on insights ever since we introduced it. We haven't done the work yet, but the plan is to move to a "store the unsaved state in the URL" approach. This way if you accidentally click away from a saved state, you can click back and then save it. Using actionToUrl with `{replace: true}` as the last parameter should make this kind of easy/possible.

Obviously feel free to ignore me :).

Reply:
I appreciate the feedback!  Neil had a similar comment about this in my implementation, so I'm going to have a whack at trying that approach instead.